lib/gprdperm.gi

```diff
@@ -813,17 +813,80 @@ local  info,proj,H;
     proj:=ActionHomomorphism(W,info.components,OnSets,"surjective");
   else
     H:=info.groups[2];
-    proj:=List(info.basegens,i->One(H));
     proj:=GroupHomomorphismByImagesNC(W,H,
-      Concatenation(info.basegens,info.hgens),
-      Concatenation(proj,GeneratorsOfGroup(H)));
+      Concatenation(Concatenation(info.basegens),info.hgens),
+      Concatenation(Concatenation(List(info.basegens, x ->
+                    ListWithIdenticalEntries(Length(x), One(H)))),
+                    GeneratorsOfGroup(H)));
   fi;
   SetKernelOfMultiplicativeGeneralMapping(proj,info.base);
 
   info.projection:=proj;
   return proj;
 end);
         
+#############################################################################
+##
+#M  ListWreathProductElementOp( <G>, <x> )
+##
+InstallMethod( ListWreathProductElementOp, "perm wreath product", true,
+ [ IsPermGroup and HasWreathProductInfo, IsObject ], 0,
+function(G, x)
+  local info, list, h, f, degK, i, constPoints, imageComponents, comp;
+
+  info := WreathProductInfo(G);
+  # Imprimitive Action, tuple (i, j) corresponds
+  # to point i + degK * (j - 1)
+  if IsBound(info.permimpr) and info.permimpr then
+    h := x ^ Projection(G);
+    f := x * Image(Embedding(G, info.degI + 1), h) ^ (-1);
+    list := ListWithIdenticalEntries(info!.degI + 1, 0);
+    for i in [1 .. info.degI] do
+        list[i] := RestrictedPermNC(f, info.components[i]) ^ info.perms[i];
+    od;
+    list[info.degI + 1] := h;
+  # Primitive Action, tuple (t_1, ..., t_degI) corresponds
+  # to point Sum_{i=1}^degI t_i * degK ^ (i - 1)
+  elif IsBound(info.productType) and info.productType then
+    h := x ^ Projection(G);
+    f := x * Image(Embedding(G, info.degI + 1), h) ^ (-1);
+    degK := NrMovedPoints(info.groups[1]);
+    # constPoints correspond to [1, 1, 1, ...], [2, 2, 2, ...], ...
+    constPoints := List([0 .. degK - 1], x -> Sum([0 .. info.degI - 1],
+                                                  y -> x * degK ^ y)) + 1;
+    # imageComponents = [ [1^f_1, 1^f_2, 1^f_3, ...],
+    #                     [2^f_1, 2^f_2, 2^f_3, ...], ... ]
+    imageComponents := List(OnTuples(constPoints, f) - 1,
+                            p -> CoefficientsQadic(p, degK) + 1);
+    for i in [1 .. degK] do
+      comp := imageComponents[i];
+      Append(comp, ListWithIdenticalEntries(info.degI - Length(comp), 1));
```

Comment:
What do you do this for? A comment would be nice.

Reply:
Fixed Points are Zeroes in the Qadic expansion. I need to append "Trailing" Zeros.

Reply:
For example if '(1, ..., 1) ^ (f_1, ..., f_m) = (1, ..., 1)`, before I append anything, I have `CoefficientsQadic(0, degK) + 1 = []`

Reply:
That makes sense. Could you add that as a comment into the source code?

Reply:
Yes, I add this as a comment

Reply:
I still find this confusing: you talk about missing zeros and that you need to append them, but then you append ones.

Reply:
You are right. I will adjust the comment again. For example the point `1` is first translated to the qadic expansion of `1 - 1 = 0 = [0, 0, ...]_degK`, then we add `1` to each integer in the expansion and get `[1, 1, ...]` as our tuple. So the problem is indeed about missing zeroes in the qadic expansion, but I append `1` because `imageComponents` already contains the translated tuples.

Reply:
I updated the comment. I hope that it is now less confusing.